lib/Doctrine/Annotations/Metadata/Constraint/EnumConstraint.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Annotations\Metadata\Constraint;
+
+use function in_array;
+
+final class EnumConstraint implements Constraint
+{
+    /** @var mixed[] */
+    private $allowedValues;
+
+    /**
+     * @param mixed[] $allowedValues
+     */
+    public function __construct(array $allowedValues)
+    {
+        $this->allowedValues = $allowedValues;
+    }
+
+    /**
+     * @param mixed $value
+     */
+    public function validate($value) : bool
+    {
+        if (! in_array($value, $this->allowedValues)) {
+            throw InvalidValue::new($this->allowedValues, $value);
```

Comment:
These conditions should be inverted so happy path returns here and unhappy path throws below (once `return true` is gone).